File: SonicXBackned/src/main/java/org/example/Server/Playlist/PlaylistCreateHandler.java
package org.example.Server.Playlist;



import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ObjectNode;
import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;
import org.example.MusicManagment.MusicFile;
import org.example.PlaylistManagment.Playlist;
import org.example.PlaylistManagement.PlaylistManager;
import org.example.UserManagment.UserService;

import java.io.IOException;
import java.io.OutputStream;
import java.sql.SQLException;

public class PlaylistCreateHandler implements HttpHandler {

    @Override
    public void handle(HttpExchange exchange) throws IOException {
        if ("POST".equals(exchange.getRequestMethod())) {
            ObjectMapper objectMapper = new ObjectMapper();


            UserService userService = UserService.getInstance();
            String requestBody = new String(exchange.getRequestBody().readAllBytes());
            ObjectNode jsonRequest = (ObjectNode) objectMapper.readTree(requestBody);
            String playlistName = jsonRequest.get("playlistName").asText();
            ObjectNode songNode = (ObjectNode) jsonRequest.get("song");

            String title = songNode.get("title").asText();
            String artist = songNode.get("artist").asText();
            String album = songNode.get("album").asText();
            int duration = songNode.get("duration").asInt();
            String path = songNode.get("path").asText();

            PlaylistManager playlistManager= null;
            try {
                playlistManager = new PlaylistManager();
            } catch (SQLException e) {
                throw new RuntimeException(e);
            }
            try {
                playlistManager.createPlaylistTableForUser(userService);
            } catch (SQLException e) {
                throw new RuntimeException(e);
            }


            MusicFile musicFile = new MusicFile(title, artist, album, duration, path);

            Playlist playlist = new Playlist(playlistName);
            playlist.addMusicFileToPlaylist(musicFile,userService);



            ObjectNode responseJson = objectMapper.createObjectNode();
            responseJson.put("success", true);

            byte[] response = responseJson.toString().getBytes();
            exchange.getResponseHeaders().add("Content-Type", "application/json");
            exchange.sendResponseHeaders(200, response.length);
            OutputStream os = exchange.getResponseBody();
            os.write(response);
            os.close();
        } else {
            exchange.sendResponseHeaders(405, -1);
        }
    }
}

File: SonicXBackendC/SonicXBackendC/Server/PlayList/PlaylistCreateHandler.cs
using Newtonsoft.Json.Linq;
using SonicXBackendC.MusicManagment;
using SonicXBackendC.PlaylistManagement;
using SonicXBackendC.PlaylistManagment;
using SonicXBackendC.UserManagement;
using System;
using System.Data.SqlClient;
using System.IO;
using System.Net;

namespace SonicXBackendC.Server.PlayList
{
    public class PlaylistCreateHandler
    {
        public void Handle(HttpListenerContext context)
        {


            if (context.Request.HttpMethod.Equals("POST", StringComparison.OrdinalIgnoreCase))
            {
                var userService = UserService.GetInstance();
                using (var reader = new StreamReader(context.Request.InputStream))
                {
                    var requestBody = reader.ReadToEnd();
                    var jsonRequest = JObject.Parse(requestBody);
                    var playlistName = jsonRequest["playlistName"].ToString();
                    var songNode = (JObject)jsonRequest["song"];

                    var title = songNode["title"].ToString();
                    var artist = songNode["artist"].ToString();
                    var album = songNode["album"].ToString();
                    var duration = songNode["duration"].Value<int>();
                    var path = songNode["path"].ToString();

                    PlaylistManager playlistManager = null;
                    try
                    {
                        playlistManager = new PlaylistManager();
                    }
                    catch (SqlException e)
                    {
                        throw new Exception("Error creating PlaylistManager", e);
                    }

                    try
                    {
                        playlistManager.CreatePlaylistTableForUser(userService);
                    }
                    catch (SqlException e)
                    {
                        throw new Exception("Error creating playlist table for user", e);
                    }

                    var musicFile = new MusicFile(title, artist, album, duration, path);

                    var playlist = new Playlist(playlistName);
                    playlist.AddMusicFileToPlaylist(musicFile, userService);

                    var responseJson = new JObject();
                    responseJson["success"] = true;

                    var response = System.Text.Encoding.UTF8.GetBytes(responseJson.ToString());
                    context.Response.ContentType = "application/json";
                    context.Response.StatusCode = (int)HttpStatusCode.OK;
                    context.Response.OutputStream.Write(response, 0, response.Length);
                }
            }
            else
            {
                context.Response.StatusCode = (int)HttpStatusCode.MethodNotAllowed;
            }
        }

    }
}
